component-builder/src/component_builder/discover.py

```diff
@@ -7,9 +7,9 @@ def get_changed(candidates, branch=None):
 
     def is_changed(candidate):
         name = branch or candidate.branch_name('stable')
-
-        b = bash('git log --oneline origin/{branch}..HEAD '
-                 '-- {0} || echo "1"'.format(candidate.path, branch=name))
+        cmd = ('git diff --shortstat origin/{branch} '
+               '-- {0} || echo "1"'.format(candidate.path, branch=name))
```

Comment:
@alexcouper The issue now with this change is that if a branch is not rebased to the reference branch, then components that have changed in the reference branch since then will be built, even if not modified in the current branch.